File: CareConnect/CareConnect.Models/Responses/ChildrenDiagnosis.cs
using System;
using System.Collections.Generic;

namespace CareConnect.Models.Responses
{
    public class ChildrenDiagnosis
    {
        public DateTime? DiagnosisDate { get; set; }

        public string? Notes { get; set; }

        public DateTime ModifiedDate { get; set; }

        public virtual Child Child { get; set; } = null!;

        public virtual Diagnosis Diagnosis { get; set; } = null!;
    }
}

File: UI/careconnect_mobile/lib/models/responses/children_diagnosis.dart
import 'package:careconnect_mobile/models/responses/child.dart';
import 'package:careconnect_mobile/models/responses/diagnosis.dart';
import 'package:json_annotation/json_annotation.dart';

part 'children_diagnosis.g.dart';

@JsonSerializable(explicitToJson: true)
class ChildrenDiagnosis {
  DateTime? diagnosisDate;
  String? notes;
  DateTime modifiedDate;
  Child child;
  Diagnosis diagnosis;

  ChildrenDiagnosis({
    this.diagnosisDate,
    this.notes,
    required this.modifiedDate,
    required this.child,
    required this.diagnosis,
  });

  factory ChildrenDiagnosis.fromJson(Map<String, dynamic> json) =>
      _$ChildrenDiagnosisFromJson(json);

  Map<String, dynamic> toJson() => _$ChildrenDiagnosisToJson(this);
}
